plugins/base/src/main/kotlin/renderers/html/innerTemplating/DefaultTemplateModelFactory.kt

```diff
@@ -91,6 +93,30 @@ public class DefaultTemplateModelFactory(
             ?: DokkaBaseConfiguration.defaultFooterMessage)
     )
 
+    private fun calculateSourceUrlFromSourceLinks(): String? {
+        val githubLinkRegex = Regex("http(s)?://github\\.com/([\\w,\\-_]+)/([\\w,\\-_]+)/.*")
```

Comment:
Optional as it's not that important atm: it looks like this function will be called for each page, so it could be optimized slightly:

1. Afaik, each instance of `Regex` will compile the pattern under the hood, which takes some time, and it's generally advised to reuse it if it makes sense and is possible. I think in our case the Regex can be extracted into a companion or a top-level val or something, to be reused.
2. We only need to calculate the return value once for the whole `DokkaConfiguration` (it will not change in-between invocations), and then we can reuse it. But I can't think of where to save this result off the top of my head. Maybe as a lateinit or a lazy property somewhere? Any ideas? :)

Reply:
If `sourceUrl` does not depend on a page, you can use `buildSharedModel`.

Reply:
Thx!
1. extracted `Regex` to companion
2. moved calculation of link to `buildSharedModel`